vent/io/devices/pins.py

```diff
@@ -155,19 +151,21 @@ def write(self, value):
         self.duty = value
 
     def __pwm(self, frequency, duty):
-        """ Description:
-        -If hardware_enabled is True, start a hardware pwm with the requested duty.
-        -Otherwise (or if setting a hardware pwm fails and hardware_enabled is write to False),
-         write a software pwm in the same manner.
+        """ Sets a PWM frequency and duty using either hardware or software generated PWM according to the value of
+        `self.hardware_enabled`. If hardware_enabled, starts a hardware pwm with the requested duty. If not
+        hardware_enabled, or if there is a problem setting a hardware generated PWM, starts a software PWM.
 
         Args:
-            frequency: A new PWM frequency to use.
-            duty (int): A pigpio integer representation of duty cycle
+            frequency (float): A new PWM frequency to use.
+            duty (float): The PWM duty cycle to set. Must be between 0 and 1.
         """
-        if self.hardware_enabled:
-            self.__hardware_pwm(frequency, duty)
-        if not self.hardware_enabled:
-            self.__software_pwm(frequency, duty)
+        if not 0 <= duty <= 1:
+            raise ValueError('Duty cycle must be between 0 and 1')

```

Comment:
for debugging, add something like 'got {}'.format(duty)

Reply:
nice thanks